src/EditorFeatures/Core/Implementation/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.TagProducer.cs

```diff
@@ -56,6 +56,10 @@ public void Dispose()
                 }
 
                 var document = snapshotSpans.First(vt => vt.SnapshotSpan.Snapshot == position.Snapshot).Document;
+                if (document == null)
+                {
+                    return SpecializedTasks.EmptyEnumerable<ITagSpan<AbstractNavigatableReferenceHighlightingTag>>();
+                }
```

Comment:
I saw an NRE here for null document, can't recall it was for additional document OR document removed from workspace. In fact, the constructor of DocumentSnapshotSpan says the Document could be null and we have a null check at all but 2 places, so making this consistent seems right.